backend/src/v5/processors/teamspaces/projects/models/drawings.js

```diff
@@ -35,7 +40,48 @@ Drawings.addDrawing = (teamspace, project, data) => addModel(teamspace, project,
 
 Drawings.updateSettings = updateModelSettings;
 
-Drawings.deleteDrawing = deleteModel;
+Drawings.deleteDrawing = async (teamspace, project, drawing) => {
+	await removeFilesWithMeta(teamspace, DRAWINGS_HISTORY_REF_COL, { model: drawing });
+
+	await Promise.all([
+		deleteModelRevisions(teamspace, project, drawing, modelTypes.DRAWING),
+		deleteModel(teamspace, project, drawing),
+		removeModelFromProject(teamspace, project, drawing),
+	]);
+};
+
+Drawings.getRevisions = async (teamspace, drawing, showVoid) => {
+	const revisions = await getRevisions(teamspace, drawing, modelTypes.DRAWING, showVoid,
+		{ _id: 1, author: 1, format: 1, timestamp: 1, statusCode: 1, revCode: 1, void: 1, desc: 1 });
+
+	return revisions;
+};
+
+Drawings.newRevision = async (teamspace, project, drawing, data, file) => {
+	const format = `.${file.originalname.split('.').splice(-1)[0].toLowerCase()}`;
+
+	const fileId = generateUUID();
+	const revId = await addRevision(teamspace, project, drawing, modelTypes.DRAWING,
+		{ ...data, format, rFile: [fileId], status: STATUSES.PROCESSING });
+
+	const fileMeta = { name: file.originalname, rid: revId, project, model: drawing };
+	await storeFile(teamspace, DRAWINGS_HISTORY_REF_COL, fileId, file.buffer, fileMeta);
+
+	await updateRevision(teamspace, drawing, modelTypes.DRAWING, revId, { status: STATUSES.OK });
```

Comment:
same here,  I don't think your update Revision fire a the right status update? have a sanity check there